seqr/utils/elasticsearch/es_utils_tests.py

```diff
@@ -1033,147 +1042,148 @@ def _set_cache(k, v):
                         for sample in var['_source'].get(sample_field, []):
                             gene_samples[sample] += 1
                     bucket[sample_field] = {'buckets': [{'key': k, 'doc_count': v} for k, v in gene_samples.items()]}
-    else:
-        del mock_response.aggregations.genes
-    return mock_response
+
+        response_dict['aggregations'] = {'genes': {'buckets': buckets}}
+
+    return response_dict
 
 
 @mock.patch('seqr.utils.redis_utils.redis.StrictRedis', lambda **kwargs: MOCK_REDIS)
-@mock.patch('seqr.utils.elasticsearch.utils.elasticsearch.Elasticsearch', lambda **kwargs: MOCK_ES_CLIENT)
-@mock.patch('seqr.utils.elasticsearch.es_search._liftover_grch38_to_grch37', lambda: MOCK_LIFTOVER)
 class EsUtilsTest(TestCase):
     fixtures = ['users', '1kg_project', 'reference_data']
     multi_db = True
 
     def setUp(self):
         Sample.objects.filter(sample_id='NA19678').update(is_active=False)
         self.families = Family.objects.filter(guid__in=['F000003_3', 'F000002_2', 'F000005_5'])
-        self.executed_search = None
-        self.searched_indices = []
 
-        def mock_execute_search(search):
-            self.executed_search = deepcopy(search.to_dict())
-            self.searched_indices += search._index
+        self.mock_es_client = mock.MagicMock()
+        self.mock_es_client.indices.get_mapping.side_effect = lambda index='': {
+            k: {'mappings': INDEX_METADATA[k]} for k in index.split(',')}
+
+        self.mock_search = self.mock_es_client.search
+        self.mock_search.side_effect = lambda index=None, body=None, **kwargs: create_mock_response(
+            deepcopy(body), index=','.join(index))
+        self.mock_multi_search = self.mock_es_client.msearch
+        self.mock_multi_search.side_effect = lambda body=None, **kwargs: {
+            'responses': [
+                create_mock_response(exec_search, index=','.join(body[i-1]['index']))
+                for i, exec_search in enumerate(deepcopy(body)) if not exec_search.get('index')]}
+
+        patcher = mock.patch('seqr.utils.elasticsearch.utils.elasticsearch.Elasticsearch')
+        patcher.start().return_value = self.mock_es_client
+        self.addCleanup(patcher.stop)
 
-            if isinstance(self.executed_search, list):
-                return [create_mock_response(exec_search, index=','.join(self.executed_search[i-1]['index']))
-                        for i, exec_search in enumerate(self.executed_search) if not exec_search.get('index')]
-            else:
-                return create_mock_response(self.executed_search, index=','.join(self.searched_indices))
+        self.mock_liftovers = {
+            'hg38': mock.MagicMock(),
+            'hg19': mock.MagicMock(),
+        }
+        self.mock_liftovers['hg38'].convert_coordinate.side_effect = lambda chrom, pos: [[chrom, pos - 10]]
+        self.mock_liftovers['hg19'].convert_coordinate.side_effect = lambda chrom, pos: [[chrom, pos + 10]]
```

Comment:
because I forgot to take those out, they are no longer used. I need these here because I can access and modify them in tests so I can test error cases

Reply:
`MOCK_LIFTOVERS` are what is returned by this mock, if you look down 3 lines